jwst/resample/resample_spec.py

```diff
@@ -275,3 +271,17 @@ def do_drizzle(self, **pars):
             self.output_models.append(output_model)
 
         return self.output_models
+
+
+def find_dispersion_axis(wavelength_array):
+    """
+    Find the dispersion axis (0-indexed) of the given 2D wavelength array
+    """
+    diffx = wavelength_array[:, 1:] - wavelength_array[:, 0:-1]
+    diffy = wavelength_array[1:, :] - wavelength_array[0:-1, :]
+    dwlx = np.abs(np.nanmean(diffx))
+    dwly = np.abs(np.nanmean(diffy))
+    if dwlx > dwly:
```

Comment:
If `dwlx` is equal to `dwly`, this function will return None.  That's probably OK, because it "shouldn't happen," and it would be caught when attempting to evaluate `spectral_axis ^ 1`; the drawback is that the error message would be somewhat obscure.

Reply:
Thanks.  Yeah, I was trying to figure out how to deal with that situation.  It would be an amazing confluence of events if that happened!  Perhaps I should catch that case and issue an error message.  `raise LotteryErrror('You win!')`

Reply:
Anyway, I'll add a `RunTimeError`.  😄